public/app/plugins/datasource/mixed/MixedDataSource.test.ts

```diff
@@ -93,6 +93,50 @@ describe('MixedDatasource', () => {
     });
   });
 
+  describe('with datasources as string', () => {
+    it('direct query should return results', async () => {
+      const ds = new MixedDatasource({} as DataSourceInstanceSettings);
+      const requestMixed = getQueryOptions({
+        targets: [
+          // @ts-ignore
+          { refId: 'QA', datasource: 'A' }, // 1
+          // @ts-ignore
+          { refId: 'QB', datasource: 'B' }, // 2
```

Comment:
where is datasource property a string? 

The dashboard schema migratory will migrate them to an object (depending on dashboard schemeVersion)